projects/grafana/grafana-ocp.yaml

```diff
@@ -0,0 +1,662 @@
+- description: The location of the proxy image
+  name: IMAGE_PROXY
+  value: openshift/oauth-proxy:v1.0.0
+- description: External URL for the grafana route
+  name: ROUTE_URL
+  value: ""
+- description: The namespace to instantiate heapster under. Defaults to 'grafana'.
+  name: NAMESPACE
+  value: grafana
+- description: The session secret for the proxy
+  name: SESSION_SECRET
+  generate: expression
+  from: "[a-zA-Z0-9]{43}"
+objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: grafana-ocp
+    namespace: "${NAMESPACE}"
+    annotations:
+      serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"grafana-ocp"}}'
+- apiVersion: authorization.openshift.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: gf-cluster-reader
+  roleRef:
+    name: cluster-reader
+  subjects:
+  - kind: ServiceAccount
+    name: grafana-ocp
+    namespace: "${NAMESPACE}"
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    name: grafana-ocp
+    namespace: "${NAMESPACE}"
+  spec:
+    host: "${ROUTE_URL}"
+    to:
+      name: grafana-ocp
+    tls:
+      termination: Reencrypt
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: grafana-ocp
+    annotations:
+      prometheus.io/scrape: "true"
+      prometheus.io/scheme: https
+      service.alpha.openshift.io/serving-cert-secret-name: gf-tls
+    namespace: "${NAMESPACE}"
+    labels:
+      metrics-infra: grafana-ocp
+      name: grafana-ocp
+  spec:
+    ports:
+    - name: grafana-ocp
+      port: 443
+      protocol: TCP
+      targetPort: 8443
+    selector:
+      app: grafana-ocp
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: gf-proxy
+    namespace: "${NAMESPACE}"
+  stringData:
+    session_secret: "${SESSION_SECRET}="
+# Deploy Prometheus behind an oauth proxy
+- apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
+    labels:
+      app: grafana-ocp
+    name: grafana-ocp
+    namespace: "${NAMESPACE}"
+  spec:
+    replicas: 1
+    selector:
+      matchLabels:
+        app: grafana-ocp
+    template:
+      metadata:
+        labels:
+          app: grafana-ocp
+        name: grafana-ocp-app
+      spec:
+        serviceAccountName: grafana-ocp
+        containers:
+        - name: oauth-proxy
+          image: ${IMAGE_PROXY}
+          imagePullPolicy: IfNotPresent
+          ports:
+          - containerPort: 8443
+            name: web
+          args:
+          - -https-address=:8443
+          - -http-address=
+          - -email-domain=*
+          - -client-id=system:serviceaccount:${NAMESPACE}:grafana-ocp
+          - -upstream=http://localhost:3000
+          - -provider=openshift
+#          - -skip-auth-regex=/api/datasources
+          - '-openshift-delegate-urls={"/api/datasources": {"resource": "namespace", "verb": "get", "resourceName": "grafana-ocp", "namespace": "${NAMESPACE}"}}'
```

Comment:
@smarterclayton @stevekuznetsov 
I use openshift-delegate-urls in order to by pass these api url
but can't find it working,

this is what I've seeing in the oauth logs
2017/12/03 14:57:44 provider.go:347: authorizer reason: User "system:anonymous" cannot get namespace in project "grafana"

any insights ?